orttraining/orttraining/training_ops/cuda/collective/nccl_kernels.cc

```diff
@@ -6,6 +6,32 @@
 namespace onnxruntime {
 namespace cuda {
 
+static Status VerifyTensorsAllocatedContiguously(OpKernelContext* context) {
```

Comment:
This is fine here - I had added it as a check while I was prototyping.  Wondering aloud - would it make sense to move it to executor before invoking the kernel (perhaps as debug-only check?)